onepassword/util/date_test.go

```diff
@@ -0,0 +1,77 @@
+package util
+
+import "testing"
+
+func TestIsValidDateFormat(t *testing.T) {
```

Comment:
According to [the documentation](https://developer.1password.com/docs/cli/item-fields/#custom-fields) the valid Date field format is `YYYY-MM-DD`. 

When trying to update an item using either CLI or Connect and providing the date in format other than `YYYY-MM-DD` it returns an error:
```
Logic: (Unable to parse Item), values for field type date must be `YYYY-MM-DD` or epoch time format
```
So, I think for now it's ok keep this validation on the provider side, to not make any accessive requests. And you're right, we need to mention that in the Provider documentation so users not to be confused.

But you raised a very good point. I think on the teraform side we can improve it to accept different types of date, parse it and sumbit a valid `YYYY-MM-DD` format on request. So, it will make it more user friendly. I'll open a separate issue for that.

Reply:
Thank you for the additional context. That makes me feel more comfortable with having the `YYYY-MM-DD` as the only supported format. I'd even go as far as not needing to add additional supported formats to keep it consistent across other integrations (e.g. using Connect SDKs, CLI, other tools).

And the plan sounds good. Do make sure it's documented and all my concerns are addressed. 😄